azure-pipelines-dev.yml

```diff
@@ -166,10 +166,14 @@ stages:
       vmImage: 'ubuntu-18.04'
 
     jobs:
-      - job: compatibility_matrix
+      - job: minimal
         condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.GEChanged'], true)
-        variables:
-          GE_pytest_opts: '--no-sqlalchemy'
+        strategy:
+          matrix:
+            standard:
+              pytest_args: 'tests --ignore "tests/rule_based_profiler"'
+            slow:
+              pytest_args: 'tests/rule_based_profiler'
```

Comment:
Rather than `pytest tests` taking a full hour, let's try and determine a strategy where we use 2 workers and have them tackle 30mins/30mins.

Thanks to refactors done on the config, we have the parallel workers to spare and have the opportunity to cut down our CI times.

Reply:
This seems like something we should use marks for.

Or better yet let's use one mark and have the standard jut be `-m 'not MY_NEW_MARK'`

Reply:
I think this can be a follow up though.

Reply:
hmm but how can we mark a full directory? I mainly split it up this way because after some testing, I was able to determine that the RBP tests make up about half our runtime

Reply:
I don't want to use `@pytest.mark.slow` since we used that previously

Reply:
I don't know how to mark a whole directory.

but to mark a whole module...

```python
import pytest

pytestmark = pytest.mark.MY_MARKER

```

https://docs.pytest.org/en/7.1.x/example/markers.html#marking-whole-classes-or-modules